compiler/src/iree/compiler/Dialect/LinalgExt/Transforms/ConvertConv2DToWinograd.cpp

```diff
@@ -291,28 +165,58 @@ class ConvertConvToWinograd final : public OpRewritePattern<ConvOp> {
   LogicalResult matchAndRewrite(ConvOp convOp,
                                 PatternRewriter &rewriter) const override {
 
-    bool isNchw;
-    if (!isValidConv2d(convOp, isNchw)) {
+    bool isNchwFchw;
+    if (!isValidConv2d(convOp, isNchwFchw)) {
       return failure();
     }
 
-    // Check that kernel has been constant folded (by validating rank = 3)
+    // Create winograd filter transform op.
     Value kernel = convOp.getInputs()[1];
     auto kernelType = cast<ShapedType>(kernel.getType());
     if (!kernelType) {
       return failure();
     }
-    Type elementType = kernelType.getElementType();
-    ArrayRef<int64_t> kernelShape = kernelType.getShape();
-    if (kernelShape.size() != 3) {
+    SmallVector<int64_t> kernelShape(kernelType.getShape());
+    if (llvm::any_of(kernelShape, ShapedType::isDynamic)) {
       return failure();
     }
+    const int64_t kh = isNchwFchw ? kernelShape[2] : kernelShape[0];
+    const int64_t kw = isNchwFchw ? kernelShape[3] : kernelShape[1];
+    if ((kh != 3) || (kw != 3)) {
+      return failure();
+    }
+    assert(kernelShape.size() == 4);
+    Type elementType = kernelType.getElementType();
 
     const int64_t kernelSize = 3;
     const int64_t inputTileSize = outputTileSize + kernelSize - 1;
 
-    // Create winograd input transform op
     Location loc = convOp.getLoc();
+    const std::array<int64_t, 2> hwcfKernelDims{0, 1};
+    const std::array<int64_t, 2> fchwKernelDims{2, 3};
+    SmallVector<int64_t> filterResultShape(4, inputTileSize);
+    filterResultShape[2] = isNchwFchw ? kernelShape[1] : kernelShape[2];
```

Comment:
I remember you had a function in one of the previous commits about querying cDim and fDim. Can you use that here?

Reply:
That is a filter transform op function, and the op needs to be created to use it (it uses some op metadata), so I can't easily use it here.